mlir/lib/ExecutionEngine/ROCm/rocm-runtime-wrappers.cpp

```diff
@@ -447,6 +649,23 @@ extern "C" void mgpuMemCopy4DBF16(
             static_cast<hipMemcpyKind>(copyDirection));
 }
 
+extern "C" void mgpuMemCopy5DBF16(
+    unsigned short *sourceAllocated, unsigned short *sourceAligned,
+    int64_t sourceOffset, int64_t sourceSize0, int64_t sourceSize1,
+    int64_t sourceSize2, int64_t sourceSize3, int64_t sourceSize4,
+    int64_t sourceStride0, int64_t sourceStride1, int64_t sourceStride2,
+    int64_t sourceStride3, int64_t sourceStride4, unsigned short *destAllocated,
+    unsigned short *destAligned, int64_t destOffset, int64_t destSize0,
+    int64_t destSize1, int64_t destSize2, int64_t destSize3, int64_t destSize4,
+    int64_t destStride0, int64_t destStride1, int64_t destStride2,
+    int64_t destStride3, int64_t destStride4, unsigned copyDirection) {
+  hipMemcpy(destAligned, sourceAligned,
+            sourceSize0 * sourceSize1 * sourceSize2 * sourceSize3 *
+                sourceSize4 * sizeof(unsigned short),
+            static_cast<hipMemcpyKind>(copyDirection));
+}
+
+typedef std::unordered_map<char, std::pair<int64_t, int64_t>> TensorDim;
 
 // Extract proper tensor sizes and strides based on layouts
 static void
```

Comment:
This function is not needed because the internal representation would be always 5-dimensions, right?